backend/src/bundles/user-details/user-details.repository.ts

```diff
@@ -31,6 +36,72 @@ class UserDetailsRepository implements Repository {
         throw new Error(ErrorMessages.NOT_IMPLEMENTED);
     }
 
+    public async searchUsers(
+        payload: UserDetailsSearchUsersRequestDto,
+    ): Promise<UserDetailsEntity[] | null> {
+        let query = this.userDetailsModel.query();
+
+        try {
+            if (payload.search) {
+                query = query.where('fullName', 'ilike', `%${payload.search}%`);
+            }
```

Comment:
You can use query in such way
```suggestion
            query.where((builder) => {
                if (payload.search) {
                    builder.where('fullName', 'ilike', `%${payload.search}%`);
                }
                ....
            })
            
```